src/NUnitFramework/tests/TestUtilities/Collections/SimpleEnumerable.cs

```diff
@@ -54,4 +54,69 @@ public IEnumerator GetEnumerator()
 
         #endregion
     }
+
+    class SimpleEnumerableWithIEquatable : IEnumerable, IEquatable<IEnumerable<object>>
```

Comment:
I'm confused why this isn't either `IEquatable<SimpleEnumerableWithIEquatable>` or `IEquatable<IEnumerable>`?
The class doesn't implement `IEnumerable<object>` so right now it is not equatable to itself.